server/src/main/java/org/elasticsearch/rest/action/RestMainAction.java

```diff
@@ -24,6 +26,7 @@
 import static org.elasticsearch.rest.RestRequest.Method.GET;
 import static org.elasticsearch.rest.RestRequest.Method.HEAD;
 
+@ServerlessScope(Scope.PUBLIC)
 public class RestMainAction extends BaseRestHandler {
```

Comment:
Are we sure the main action is in scope for serverless? In its current form it exposes a bunch of information that will not make sense in serverless: cluster name, cluster uuid, build information (which includes the self managed version number). Maybe this should be internal to start?

Reply:
Oh hmm good point. I'll change it to internal.

Reply:
Done.